rewrite-java/src/main/java/org/openrewrite/java/cleanup/AutoCleanup.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Incubating;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.tree.J;
+
+@Incubating(since = "7.0.0")
+public class AutoCleanup extends Recipe {
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new AutoCleanupFromCompilationUnit();
+    }
+
+    private static class AutoCleanupFromCompilationUnit extends JavaIsoVisitor<ExecutionContext> {
+        @Override
+        public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext context) {
+            doAfterVisit(new AutoCleanupVisitor<>());
+            return cu;
+        }
+    }
```

Comment:
What does the existence of `AutoCleanupFromCompilationUnit` accomplish? Why doesn't `AutoCleanup.getVisitor()` return `new AutoCleanupVisitor<>()` directly?
Might just be some aspect of how 7.x works that I'm not understanding; I'm still looking at things with a substantially 6.x-oriented perspective.

Reply:
~~So basically what it's doing is acting as an entrypoint for chaining multiple `cleanup`-related recipes together through the `compilationUnit` visit as the element to call and start the `AutoCleanup()` chain, starting on the _root_ compilation unit to ensure styles are--~~ 

You know what, as I write this, I'm not able to give you as good of an explanation as I thought I understood the other evening. I'm going to make changes & make a note that to better understand `AutoCleanupFromCompilationUnit`'s use in AutoFormat

Reply:
(Decided to remove AutoCleanup visitor for now since there's not really a reason it _has_ to be included in this PR with this visitor)